src/kernels/MIOpenKthvalue.cpp

```diff
@@ -0,0 +1,204 @@
+     */
+
+    size_t lid = threadIdx.x;
+    size_t gid = blockIdx.x;
+    if(gid >= output_size)
+    {
+        return;
+    }
+
+    __shared__ size_t lsum[LOCAL_SIZE][RADIX_SIZE];
+    __shared__ DTYPE lval;
+    __shared__ long lidx;
+    size_t counts[RADIX_SIZE];
+    RADIX_TYPE desired_mask = 0;
+    RADIX_TYPE desired      = 0;
+
+    tensor_layout_t<4> layout(input_tv, gid);
+    auto idx = input_tv.get_tensor_view_idx(layout);
+
+    for(int pos = sizeof(RADIX_TYPE) * 8 - RADIX_BITS; pos >= 0; pos -= RADIX_BITS)
+    {
+        for(unsigned long& count : counts)
+        {
+            count = 0;
+        }
+
+        for(size_t i = lid; i < dim_size; i += LOCAL_SIZE)
+        {
+            size_t input_idx = idx + i * dim_stride;
+            RADIX_TYPE val   = encode<DTYPE>(input[input_idx]);
+            if((val & desired_mask) == desired)
+            {
+                ++counts[GetBitFieldImpl<RADIX_TYPE>(val, pos, RADIX_BITS)];
+            }
+        }
+
+        for(size_t i = 0; i < RADIX_SIZE; ++i)
+        {
+            lsum[lid][i] = counts[i];
+        }
+        __syncthreads();
+        for(size_t i = LOCAL_SIZE >> 1; i > 0; i >>= 1)
+        {
+            if(lid < i)
+            {
+                for(size_t j = 0; j < RADIX_SIZE; ++j)
+                {
+                    lsum[lid][j] += lsum[lid + i][j];
+                }
+            }
+            __syncthreads();
+        }
+        for(size_t i = 0; i < RADIX_SIZE; ++i)
+        {
+            counts[i] = lsum[0][i];
+        }
+        __syncthreads();
+
+        bool found = false;
+        // Process in ascending order
+        for(size_t j = 0; j < RADIX_SIZE; ++j)
+        {
+            if(counts[j] < k)
+            {
+                k -= counts[j];
+                continue;
+            }
+            // Answer is inside this count
+            if(counts[j] == 1 || pos == 0)
+            {
+                // 1. counts[j] == 1
+                // We found an unique answer.
+                // 2. pos == 0
+                // There are multiple answers so we return any of them
+                for(size_t i = lid; i < dim_size; i += LOCAL_SIZE)
+                {
+                    size_t input_idx = idx + i * dim_stride;
+                    DTYPE val_ori    = input[input_idx];
+                    RADIX_TYPE val   = encode<DTYPE>(val_ori);
+                    if((val & desired_mask) == desired &&
+                       GetBitFieldImpl<RADIX_TYPE>(val, pos, RADIX_BITS) == j)
+                    {
+                        // For case 2, this will be non-deterministic.
+                        lval = val_ori;
+                        lidx = i;
+                    }
+                }
+                found = true;
+                break;
+            }
+            desired      = SetBitFieldImpl<RADIX_TYPE>(desired, j, pos);
+            desired_mask = SetBitFieldImpl<RADIX_TYPE>(desired_mask, RADIX_MASK, pos);
+            break;
```

Comment:
This `break` changes the loop at the line 135 from
`for(size_t j = 0; j < RADIX_SIZE; ++j)`
to
`for(size_t j = 0; j < 1; ++j)`

Reply:
It's not, in the begining of the loop, I have this if block to continue iterating through the loop.